File: Java/1036.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
public class Main {
    public static void main(String[] args) throws IOException {
        InputStreamReader ir = new InputStreamReader(System.in);
        BufferedReader in = new BufferedReader(ir);
        double a,b,c,x1,x2;
        String line=in.readLine();
        String[] values= line.trim().split(" ");
        a=Double.parseDouble(values[0]);
        b=Double.parseDouble(values[1]);
        c=Double.parseDouble(values[2]);
        double delta = Math.pow(b,2) - 4 * a * c;
        if(a == 0.0 || delta < 0){
            System.out.printf("Impossivel calcular\n");
        }else{
            x1=(-b + Math.sqrt(delta)) / (2*a);
            x2=(-b - Math.sqrt(delta)) / (2*a);
            System.out.printf("R1 = %.5f\n",x1);
            System.out.printf("R2 = %.5f\n",x2);
        }
    }
}

File: C++/1036.cpp
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

class Bhaskara {
    private:
        double a;
        double b;
        double c;
    public:
        void ler();
        int Eq2Grau();
        double delta();
        double raiz1();
        double raiz2();
};

void Bhaskara::ler() {
    cin >> a >> b >> c;
}

double Bhaskara::delta() {
    float d = pow(b, 2) - 4 * a * c;
    return d;
}

int Bhaskara::Eq2Grau() {
    if(a == 0.0) {
        return 0;
    }
    return 1;
}

double Bhaskara::raiz1() {
    float r1 = (-b + sqrt(delta())) / (2 * a);
    return r1;
}

double Bhaskara::raiz2() {
    float r2 = (-b - sqrt(delta())) / (2 * a);
    return r2;
}

int main() {

	Bhaskara B;
	
	B.ler();

	if (B.delta() < 0 || !B.Eq2Grau())
		cout << "Impossivel calcular" << endl;

	else {
		cout << fixed << setprecision(5);
		cout << "R1 = " << B.raiz1() << endl;
		cout << "R2 = " << B.raiz2() << endl;
	}
	return 0;
}